### ISSUE TITLE
Cross entropy error

### ISSUE THREAD

**Issue by racevictor** [Artelnics/OpenNN#31]
In cross_entropy_error.cpp, the outputs are calculated on multilayer_perceptron but not neural_network. So the loss function does not incorporate the probability layer and will introduce a log(-ve) situation.

This problem occurs in ALL error classes. The evaluated loss functions do not include the probability layer, conditional layer etc.

**Comment by racevictor** [comment]
To reproduce:
Modify the iris plant example, use 4:6:3 architecture
(In fact it is a 3 class classification problem)
Set the loss_index to cross entropy

**Comment by shlublu** [comment]
I've seen that too. I get cross entropy losses = NaN with one hot vectors targets despite the softmax output layer the NN is assigned. 

Calling `neural_network.calculate_outputs(v);` gives correct results though, the issue is internal: `CrossEntropyError::calculate_error()` calls `multilayer_perceptron_pointer->calculate_outputs(inputs);` instead of `neural_network_pointer->calculate_outputs(inputs);`, which skips the softmax layer and makes `cross_entropy_error -= (targets[j]*log(outputs[j]) + (1.0 - targets[j])*log(1.0 - outputs[j]));` to return NaN.
